drift/core/kltransform.py

```diff
@@ -391,18 +399,26 @@ def transform_save(self, mi):
         evalsf = np.zeros(nside, dtype=np.float64)
         if evals.size != 0:
             evalsf[(-evals.size) :] = evals
+
         f.create_dataset("evals_full", data=evalsf)
 
         # Discard eigenmodes with S/N below threshold if requested.
         if self.subset:
             i_ev = np.searchsorted(evals, self.threshold)
-
-            evals = evals[i_ev:]
-            evecs = evecs[i_ev:]
-            print(
-                "Modes with S/N > %f: %i of %i"
-                % (self.threshold, evals.size, evalsf.size)
-            )
+            if self.diagonalisation_order == "sf" or self._dkl:
```

Comment:
A comment explaining the DKL part would be helpful here. If I understand correctly, the second step in the DKL will be S/N regardless of what is done for the first step, so we still want to save the DKL values in S/N order - it would be good to write that in a comment